multidict/_pair_list.c

```diff
@@ -188,6 +269,40 @@ pair_list_add_with_hash(PyObject *op,
 }
 
 
+int
+pair_list_add(PyObject *op,
+              PyObject *key,
+              PyObject *value)
+{
+    Py_hash_t hash;
+    PyObject *identity = NULL;
+    PyObject *str_key = NULL;
+    pair_list_t *list = (pair_list_t *)op;
+    int ret;
+
+    identity = list->calc_identity(key);
```

Comment:
Indirect call by function ptr can be issued for performance optimization. We should understand this. Potentially we can use a preprocessor, but it will be more dirty code.

Reply:
I understand it but I think the code is ok.
It's like a virtual function call, very widely used technic by Python Core itself.